src/mscorlib/src/System/IO/Stream.cs

```diff
@@ -116,7 +116,37 @@ internal SemaphoreSlim EnsureAsyncActiveSemaphoreInitialized()
         [ComVisible(false)]
         public Task CopyToAsync(Stream destination)
         {
-            return CopyToAsync(destination, _DefaultCopyBufferSize);
+            int bufferSize = _DefaultCopyBufferSize;
+            if (CanSeek)
```

Comment:
Could you please put the optimization under `#if FEATURE_CORECLR`? We will consider it for full framework separately.